discover/deployment/articles-dxp/05-maintaining-liferay/01-patching-liferay-dxp/00-patching-liferay-dxp-intro.markdown

```diff
@@ -10,8 +10,14 @@ hotfixes.
 
 **Important:** Make sure to
 [back up your @product@ installation and database](/discover/deployment/-/knowledge_base/7-1/backing-up-a-liferay-installation)
-regularly, especially before patching. Patches can contain @product@ Core data
-changes---the patching tool applies them automatically. 
+regularly, especially before patching. Certain fix packs (service packs) can
+contain data changes in @product@ modules (micro changes in their schema
+version which are revertibles) which are applied by the patching tool
+automatically. From 7.1 also micro data changes in @product@ Core can be
+included on fix packs and they are able to be executed on demand using the db
+upgrade tool. Please, review the release notes and the documentation for the
```

Comment:
@achaparro From 7.1 are micro data changes executed by the patching tool automatically? If yes, in what cases for DXP would you want to run the db upgrade tool before running the patching tool?


Thanks!

Reply:
Hi @jhinkey 

Data changes are never executed by the patching tool, the patching tool installs the new code in the server/OSGi engine and new upgrades are executed automatically on server startup. For Core we can, optionally, run new upgrade processes (if there are new ones) using the db upgrade tool before starting up the server. Please take into account that we can't execute them on server startup because we don't want to delay this process. They are just micro changes so they are not mandatory so they server can also start up if we don't execute the db upgrade tool.

Let me know if you have any question about it.

Thanks.